uPortal-webapp/src/test/java/org/apereo/portal/portlet/dao/jpa/JpaMarketplaceRatingDaoTest.java

```diff
@@ -279,4 +286,137 @@ public Object call() throws Exception {
+                        assertEquals(ratingList.size(), 1);
+
+                        daysBack = 30;
+                        ratingList = marketplaceRatingDao.getAllRatingsInLastXDays(daysBack);
+                        assertEquals(ratingList.size(), 2);
+
+                        daysBack = 5000;
+                        ratingList = marketplaceRatingDao.getAllRatingsInLastXDays(daysBack);
+                        assertEquals(ratingList.size(), 4);
+
+                        return null;
+                    }
+                });
+    }
+
+    @Test
+    public void testGetAllRatingsForPortletInLastXDays() {
+        // Let's create some ratings
+        this.execute(
+                new Callable<Object>() {
+                    @Override
+                    public Object call() throws Exception {
+
+                        Calendar today = Calendar.getInstance();
+
+                        List<IPortletDefinition> portletList =
+                                portletDefinitionDao.getPortletDefinitions();
+                        List<ILocalAccountPerson> personList = localAccountDao.getAllAccounts();
+                        assertNotEquals(portletList.size(), 0);
+                        assertNotEquals(personList.size(), 0);
+                        int i = 1;
+                        for (IPortletDefinition portlet : portletList) {
+
+                            for (ILocalAccountPerson person : personList) {
+                                today.add(Calendar.DAY_OF_YEAR, 0 - i);
+                                System.out.println(
+                                        "Testing get ratings using "
+                                                + today.getTime()
+                                                + " for portlet creation date. Porlet is "
+                                                + portlet.getFName());
+                                MarketplaceRatingImpl rating = new MarketplaceRatingImpl();
+                                int starRating =
+                                        random.nextInt(MarketplaceRatingImpl.MAX_RATING) + 1;
+                                rating.setRating(starRating);
+                                MarketplaceRatingPK ratingPK = new MarketplaceRatingPK();
+                                ratingPK.setPortletDefinition((PortletDefinitionImpl) portlet);
+                                ratingPK.setUserName(person.getName());
+                                rating.setMarketplaceRatingPK(ratingPK);
+                                rating.setRatingDate(today.getTime());
+                                rating.setReview(
+                                        "comment for "
+                                                + ratingPK.getPortletDefinition().getFName());
+                                marketplaceRatingDao.createOrUpdateRating(rating);
+                                i = i + 20;
```

Comment:
:question: Could `i` be named something `numberOfDaysBack` to give some additional clarity on the purpose?
:question:  maybe `20` could become something like `final int TEST_DATE_INCREMENT = 20`?

continued from https://github.com/Jasig/uPortal/pull/1064#discussion_r155368288

Reply:
made this change

Reply:
Thanks @phillips1021!Ccould the change also be pushed to GitHub?

Reply:
It was pushed to GitHub previously - see: https://github.com/phillips1021/uPortal/commit/d6764ce4044c9369e5ce8bc227279017c0cbf7a4

Reply:
Line 395 uses the same constant for the same purpose.